pkg/prometheus/statefulset.go

```diff
@@ -616,10 +616,14 @@ func makeStatefulSetSpec(p monitoringv1.Prometheus, c *operator.Config, shard in
 	podAnnotations := map[string]string{}
 	podLabels := map[string]string{}
 	podSelectorLabels := map[string]string{
-		"app":                   "prometheus",
-		"prometheus":            p.Name,
-		shardLabelName:          fmt.Sprintf("%d", shard),
-		prometheusNameLabelName: p.Name,
+		"app":                          "prometheus",
+		"app.kubernetes.io/name":       "prometheus",
+		"app.kubernetes.io/version":    version.String(),
```

Comment:
This change causes prometheus-operator to delete and recreate the statefulset on prometheus version upgrades. Previously behavior was a graceful rolling deployment. Not ideal for users with long startup times.

Reply:
Good catch, I hadn't thought about this. Do you want to create a PR that moves it out of the selectors into the pod labels? Otherwise I can take a look in the next couple of days.

Reply:
Yup, i can do that.

Reply:
Good catch! I created an issue to track this in https://github.com/prometheus-operator/prometheus-operator/issues/4076